cupy/_core/_fusion_variable.pyx

```diff
@@ -188,9 +198,10 @@ class _TraceScalar(_TraceVariable):
         if self.dtype.kind == 'c':
             return '{}({}, {})'.format(
                 get_typename(self.dtype),
-                self.const_value.real,
-                self.const_value.imag)
-        return str(self.const_value)
+                self._float_to_str(self.const_value.real),
+                self._float_to_str(self.const_value.imag),
+            )
+        return self._float_to_str(self.const_value)
```

Comment:
Can't this whole function/property just call:
```
from cupy._core._scalar cimport get_typename, _get_cuda_scalar_repr

_get_cuda_scalar_repr(self.const_value, self.dtype)
```
It seems to change `complex<double>` to `thrust::complex<double>` but that seems fine and already the case in the other code path also?

(As is, it would still fail NaN while the other function is tested in the other path.)

Reply:
Thank you for your review! I have updated it to use `_get_cuda_scalar_repr`.

`using thrust::complex;` is declared in `complex.cuh`, so `complex<double>` is equivalent to `thrust::complex<double>`. However, I have rewritten it as `thrust::complex` to prevent confusion.

https://github.com/cupy/cupy/blob/34e2daa7c36c44bd34a2732d5343619f9850c96c/cupy/_core/include/cupy/complex.cuh#L5